code/controllers/Processes/spacedrift.dm

```diff
@@ -0,0 +1,57 @@
+#define MAX_THROWING_DIST 512 // 2 z-levels on default width
+#define MAX_TICKS_TO_MAKE_UP 3 //how many missed ticks will we attempt to make up for this run.
+
+var/global/datum/controller/process/spacedrift/drift_master
+
+/datum/controller/process/spacedrift
+	var/list/processing_list = list()
+
+/datum/controller/process/spacedrift/setup()
+	name = "spacedrift"
+	schedule_interval = 5
+	start_delay = 20
+	log_startup_progress("Spacedrift starting up.")
+	if(drift_master)
+		qdel(drift_master)
+	drift_master = src
+
+/datum/controller/process/spacedrift/statProcess()
+	..()
+	stat(null, "P:[processing_list.len]")
+
+/datum/controller/process/spacedrift/doWork()
+	var/list/currentrun = processing_list.Copy()
+
+	while(currentrun.len)
```

Comment:
TG does this and the whole `var/list/currentrun = processing_list.Copy()` thing, I think, as a way to gracefully recover from runtimes/lock ups in the processor; I dunno how much use they are to us.

I think we could get away with doing this:

```
for(var/thing in processing_list)
	var/atom/movable/AM = thing)
```
Then doing the same thing.

I'm not sure it'll make a huge difference in performance though, and with the scheduler going the way of the dinosaur, I suppose this would make the transition easier?